crates/bindings-csharp/BSATN.Runtime/BSATN/Runtime.cs

```diff
@@ -403,8 +403,18 @@ public AlgebraicType GetAlgebraicType(ITypeRegistrar registrar) =>
     where ElementRW : IReadWrite<Element>, new()
 {
     private static readonly Enumerable<Element, ElementRW> enumerable = new();
+    private static readonly ElementRW elementRW = new();
 
-    public Element[] Read(BinaryReader reader) => enumerable.Read(reader).ToArray();
+    public Element[] Read(BinaryReader reader)
+    {
+        var count = reader.ReadInt32();
```

Comment:
Enumerable also uses ReadInt32, although I'm slightly confused why it isn't ReadUint32.